homeassistant/components/backup/__init__.py

```diff
@@ -7,7 +7,12 @@
 
 from .const import DOMAIN, LOGGER
 from .http import async_register_http_views
-from .manager import BackupManager
+from .manager import (
+    Backup,  # noqa: F401
+    BackupManager,
+    BackupSyncAgent,  # noqa: F401
+    SyncedBackup,  # noqa: F401
+)
```

Comment:
Could we replace `F401` with `__all__`?

Reply:
Is that what we are doing now?
Can you point me to the dev documentation for that?
<img width="609" alt="image" src="https://github.com/user-attachments/assets/2a6fb6a5-0567-44a3-8d43-fe30d0d9e4da">

Reply:
Nothing on documentation
I think that applies to core only (where constants/functions/etc. are used by other integrations)
See #84973
See https://github.com/home-assistant/core/blob/4d04402ad4473a95b300b9c9503e0daabb261ae1/homeassistant/components/sensor/__init__.py#L96-L111

Reply:
And from that same file, see: https://github.com/home-assistant/core/blob/4d04402ad4473a95b300b9c9503e0daabb261ae1/homeassistant/components/sensor/__init__.py#L67

Reply:
The purpose is different:
- `__all__` is what we want to make available for other integrations
- `F401` is for obsolete "re-exports" to avoid a breaking change